### DISCUSSION TITLE
AUDIOKIT : player.stop,play,previous,next : Am i able to use it without audiokit buttons?

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [pschatzmann/arduino-audio-tools#1558] [Q&A] [answered]
I try to use these functions touching buttons on my tft,but it doesn't respond and stop playing, with a little freeze-sound.
With the audiokit buttons all work fine,are these player functions exlusive to process actions,and by extension only
for audiokit?

**Comment by pschatzmann** [comment]
I don't understand you: What do you mean with touching buttons on my tft ?
What hardware you you using ? 
What kind of logic/functionality do you want to implement ? 
Are the pins active low or high ?

**Comment by TangerineDreamer** [comment]
I don't use the audiokit buttons and their pins,all my controls come from a tft touch wired with SPI on the audiokit
ex:
if(button next==true)
player.next();

**Comment by pschatzmann** [comment, marked as answer]
So you just need to use some event handling functionality of a TFT library. 
The AudioTools is an audio library and does not provide any TFT integration

**Comment by TangerineDreamer** [comment]
I can use my tft touch selecting a song and enter a choice like:
player.setPath(choice); that works,but not with player.next ()or player.previous() or player.stop()
So i have the tft functionnality working with audiotools

**Reply by pschatzmann** [reply]
Maybe you just need to denounce the input ? 
Did you double check that you are not calling next() many times when you just press the button ?

**Comment by TangerineDreamer** [comment]
It could be a solution,or maybe an explication because if the "process action with buttons and pins" logic take care about
debouncing and my tft logic button don't,maybe if i override the "process action with buttons and pins" logic,i could simply
enter in my logic:
rubbish code: -> if(button next==true & pin(4)==HIGH)
player.next();

**Comment by TangerineDreamer** [comment]
i made a mistake,i should write like (rubbish): if(button_next==true)) {pin(4)=HIGH}
and so go inside the "process action with buttons and pins" logic

**Comment by pschatzmann** [comment]
There is a Debouncer in the AudioTools and there are also a couple of examples.
Otherwise: Google is your friend...

**Comment by TangerineDreamer** [comment]
Ok,i will look into the library and google,thank you
